src/main/java/io/reactivex/internal/operators/observable/ObservableReduceSeedSingle.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.operators.observable;
+
+import io.reactivex.*;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.exceptions.Exceptions;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.internal.disposables.DisposableHelper;
+import io.reactivex.internal.functions.ObjectHelper;
+import io.reactivex.plugins.RxJavaPlugins;
+
+/**
+ * Reduce a sequence of values, starting from a seed value and by using
+ * an accumulator function and return the last accumulated value.
+ *
+ * @param <T> the source value type
+ * @param <R> the accumulated result type
+ */
+public final class ObservableReduceSeedSingle<T, R> extends Single<R> {
+
+    final ObservableSource<T> source;
+
+    final R seed;
+
+    final BiFunction<R, ? super T, R> reducer;
+
+    public ObservableReduceSeedSingle(ObservableSource<T> source, R seed, BiFunction<R, ? super T, R> reducer) {
+        this.source = source;
+        this.seed = seed;
+        this.reducer = reducer;
+    }
+
+    @Override
+    protected void subscribeActual(SingleObserver<? super R> observer) {
+        source.subscribe(new ReduceSeedObserver<T, R>(observer, reducer, seed));
+    }
+
+    static final class ReduceSeedObserver<T, R> implements Observer<T>, Disposable {
+
+        final SingleObserver<? super R> actual;
+
+        final BiFunction<R, ? super T, R> reducer;
+
+        R value;
+
+        Disposable d;
+
+        public ReduceSeedObserver(SingleObserver<? super R> actual, BiFunction<R, ? super T, R> reducer, R value) {
+            this.actual = actual;
+            this.value = value;
+            this.reducer = reducer;
+        }
+
+        @Override
+        public void onSubscribe(Disposable d) {
+            if (DisposableHelper.validate(this.d, d)) {
+                this.d = d;
+
+                actual.onSubscribe(this);
+            }
+        }
+
+        @Override
+        public void onNext(T value) {
+            R v = this.value;
```

Comment:
No need for separate done flag as this.value can't be initially null and becomes null if onError is called. Updated the PR with more checks.